planner/core/task.go

```diff
@@ -835,6 +835,8 @@ func (p *PhysicalLimit) attach2Task(tasks ...task) task {
 			stats := deriveLimitStats(childProfile, float64(newCount))
 			pushedDownLimit := PhysicalLimit{Count: newCount}.Init(p.ctx, stats, p.blockOffset)
 			cop = attachPlan2Task(pushedDownLimit, cop).(*copTask)
+			// Don't use clone() so that Limit and its children share the same schema. Otherwise the virtual generated column may not be resolved right.
+			pushedDownLimit.SetSchema(pushedDownLimit.children[0].Schema())
```

Comment:
`PhysicalLimit` inherits its child schema in the old implementation. I think the problem is that `ExpandVirtualColumns` of `TableScan` happens **AFTER** inheritance.

Inheritance: 
https://github.com/pingcap/tidb/blob/369c12f779f8e990f38af6a158be1d702aef4922/planner/core/task.go#L718

ExpandVirtualColumn:
https://github.com/pingcap/tidb/blob/369c12f779f8e990f38af6a158be1d702aef4922/planner/core/task.go#L730

Reply:
ExpandVirtualColumns of TableScan happens AFTER inheritance.

It is. So let them share the same schema.